Fix/media bottom sheet

baraa0abd opened the pull request:
<img width="267" height="251" alt="image" src="https://github.com/user-attachments/assets/0186c89b-6278-4da3-9523-cba81622ec0b" />

<img width="1358" height="302" alt="image" src="https://github.com/user-attachments/assets/a241c3ae-f7d2-43f8-a858-17f60daaef6e" />


- fix the media bottoms color (no more green or yellow colors in add to list and rate bottoms)
- fix the place of the movie poster (bottom padding)

baraa0abd committed:
feat: Introduce WatchListItemUiState for clearer list display

This commit introduces a new data class `WatchListItemUiState` to better represent watchlist items in the UI.

The `MovieListViewModel` has been updated to:
- Utilize `GetWatchListsUseCase` directly.
- Map the fetched `WatchList` domain objects to `WatchListItemUiState` for UI consumption.
- Populate `watchListItems` in the `MovieListUiState` with these mapped objects.

baraa0abd committed:
feat: Introduce WatchListItemUiState for clearer list display

This commit introduces a new data class `WatchListItemUiState` to better represent watchlist items in the UI.

The `MovieListViewModel` has been updated to:
- Utilize `GetWatchListsUseCase` directly.
- Map the fetched `WatchList` domain objects to `WatchListItemUiState` for UI consumption.
- Populate `watchListItems` in the `MovieListUiState` with these mapped objects.

baraa0abd committed:
fix: Adjust MoviePosterDetailScreen padding and image size

baraa0abd committed:
Merge branch 'develop' into fix/media-bottom-sheet

ahmedhgabr commented on presentation/src/main/java/com/madrid/presentation/component/movieActorBackground/MoviePosterDetailScreen.kt:
```diff
@@ -54,11 +54,10 @@ fun MoviePosterDetailScreen(
-                        .padding(bottom = 16.dp)
+                        .padding(top = 88.dp, bottom = 24.dp , start = 80.dp, end = 80.dp)
                         .size(width = 200.dp, height = 260.dp)
-                        .align(Alignment.BottomCenter)
```
The image is shifted to the left. The padding didn't solve it.

baraa0abd committed:
Merge branch 'develop' into fix/media-bottom-sheet

baraa0abd committed:
Merge branch 'develop' into fix/media-bottom-sheet

baraa0abd committed:
fix: Adjust padding in MoviePosterDetailScreen

The start padding of the Box composable within MoviePosterDetailScreen was changed from 80.dp to 90.dp.

baraa0abd committed:
Merge branch 'develop' into fix/media-bottom-sheet

Alaakhaled101 commented on presentation/src/main/java/com/madrid/presentation/component/movieActorBackground/MoviePosterDetailScreen.kt:
```diff
@@ -54,11 +54,10 @@ fun MoviePosterDetailScreen(
                 Box(
                     modifier = Modifier
-                        .padding(bottom = 16.dp)
+                        .padding(top = 88.dp, bottom = 24.dp , start = 90.dp, end = 80.dp)
```
the fixed padding in start and end is not correct it should be:
modifier = Modifier.align(Alignment.BottomCenter)
To ensure the card is in the middle while changing the screen width.

![WhatsApp Image 2025-08-16 at 19 26 27_16238d55](https://github.com/user-attachments/assets/f61d8526-1873-4725-9cd1-fca0ccafb649)

baraa0abd committed:
Merge branch 'develop' into fix/media-bottom-sheet

baraa0abd committed:
Refactor: Center movie poster and adjust padding

The movie poster in `MoviePosterDetailScreen` is now centered horizontally using `Alignment.BottomCenter`.
The vertical padding has also been adjusted, with the top padding removed and the bottom padding changed to `35.dp`.

baraa0abd committed:
Merge branch 'develop' into fix/media-bottom-sheet
